Providers/Scripts/2.4x-2.5x/Scripts/nxOMSGenerateInventoryMof.py

```diff
@@ -0,0 +1,167 @@
+    if Instances is not None :
+        for instance in Instances:
+            if instance['InstanceName'].value is not None:
+                instance['InstanceName']=instance['InstanceName'].value
+
+            if instance['ClassName'].value is not None:
+                instance['ClassName']=instance['ClassName'].value
+
+            new_properties = []
+            if instance['Properties'] is not None and len(instance['Properties'].value) > 0:
+                for property in instance['Properties'].value:
+                    if property is not None and len(property) > 0:
+                        new_properties.append(property)
+
+            if len(new_properties) > 0:
+                instance['Properties'] = new_properties
+                new_instances.append(instance)
+
+    Instances = new_instances
+    
+def Set_Marshall(FileName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    init_vars(Instances)
+    Set(FileName, Enable, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+    return [0]
+
+def Test_Marshall(FileName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    init_vars(Instances)
+    return Test(Enable, Instances)
+
+def Get_Marshall(FileName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    arg_names = list(locals().keys())
+    init_vars(Instances)
+    
+    CurrentInstances = Instances
+    FileName = protocol.MI_String(FileName)
+    Enable = protocol.MI_Boolean(Enable)
+    for instance in CurrentInstances:
+        instance['InstanceName'] = protocol.MI_String(instance['InstanceName'])
+        instance['ClassName'] = protocol.MI_String(instance['ClassName'])
+        if instance['Properties'] is not None and len(instance['Properties']):
+            instance['Properties'] = protocol.MI_StringA(instance['Properties'])
+    Instances = protocol.MI_InstanceA(CurrentInstances)
+    RunIntervalInSeconds = protocol.MI_Uint64(RunIntervalInSeconds)
+    Tag = protocol.MI_String(Tag)
+    Format = protocol.MI_String(Format)
+    FilterType = protocol.MI_String(FilterType)
+
+    if Configuration is None:
+        Configuration = []
+    if Configuration is not None and len(Configuration):
+        Configuration = protocol.MI_StringA(Configuration)
+
+    retd = {}
+    ld = locals()
+    for k in arg_names:
+        retd[k] = ld[k]
+    return 0, retd
+            
+def Set(FileName, Enable, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration):
+    if Enable == True:
+         GenerateInventoyMOF(FileName, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+    return [0]
+
+def Test(Enable, Instances):
+    return [-1]
```

Comment:
@robbiezhang The initial thinking was to have the logic to match the hash of contents (of the conf and the inventory MOF file) with the created file from config Mof in the test method. If it fails then return -1 and then again do the creation in the set(). @gauravga's argument was that as this work is duplicated in both set() and test(), it would be better to make the set() be called every time and make the work of hash match happen only once.

I do agree with the argument. We can add some smaller, less compute intensive tasks in test such as file check, etc. but it'll still have to return -1 for the actual hash match.